README.md

```diff
@@ -554,6 +554,49 @@ state = State(
 )
 ```
 
+# Actions
+
+How to test actions with scenario:
+
+```python
+from scenario import Action, Context, State
+from charm import MyCharm
+
+def test_backup_action():
+    # define an action
+    action = Action('do_backup')
+    ctx = Context(MyCharm)
+
+    # obtain an Event from the action and fire it in the context on a state of your choosing.
+    
+    # If you didn't declare do_backup in the charm's `actions.yaml`, the `ConsistencyChecker` will slap you on both wrists 
+    # and refuse to proceed.
+    ctx.run(action.event, State())
+    
+    # you can assert action results, logs, failure using the action.output interface
+    assert action.output.results == {'foo': 'bar'}
```

Comment:
A couple of thoughts:

1) It seems like this is conflating regular events with actions. I know they're sort of conflated in ops anyway (you `Framework.observe` an action) but I wonder if it would be clearer to separate them here?
2) It seems odd to me that output is stuffed onto `action.output.X`.

Brainstorming a bit, but what about this API?

```
def test_backup_action():
    ctx = scenario.Context(MyCharm)
    output, out_state = ctx.run_action('do_backup', State())
    # "output" is an ActionOutput dataclass with .results, .logs, etc
    assert output.results == {'foo': 'bar'}
    ...
```

Or maybe actions need params, which case you could pass an `Action` instance to `run_action` as well.

Reply:
Mmm, interesting. I was indeed assuming we'd want to conflate regular events with actions. 
What is specifically odd in your opinion about `action.output.X`?

Is `action.get_output().X` better? Which would raise an exception if the context hasn't run the action yet?

I don't quite mind adding a separate `ctx.run_action` but for the fact that the internal implementation would be exactly the same, so it would be an alias for `run`. 
As for separately returning the output as a second value next to the output state, API-wise I almost never like it when a function call returns a tuple. I'd rather have `state_out, action_result = ctx.run_action(action), ctx.get_action_result(action)` but that is also not very nice.

And yes, actions take params, which is why one way or another we can't pass the 'event name' only to `ctx.run`, but we need to attach the Action (with params) to the Event, somehow.